ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor.hs

```diff
@@ -667,3 +766,34 @@ wakeupDecision st _now =
     decisionState = st,
     decisionJobs  = []
   }
+
+
+-- | Classify if a node is in promiscuous mode.
+--
+-- A node is not in promiscuous mode only if: it doesn't use ledger peers, peer
+-- sharing, the set of bootstrap peers is empty.
+--
+readPromiscuousMode
+  :: MonadSTM m
+  => STM m UseLedgerPeers
+  -> PeerSharing
+  -> UseBootstrapPeers
+  -> PeerSelectionSetsWithSizes peeraddr
+  -> STM m PromiscuousMode
+readPromiscuousMode
+  readUseLedgerPeers
+  peerSharing
+  useBootstrapPeers
+  PeerSelectionView {
+    viewKnownBootstrapPeers = (knownBootstrapPeers, _)
+  }
+  =
+  do useLedgerPeers <- readUseLedgerPeers
+     pure $
+       case (useLedgerPeers, peerSharing, useBootstrapPeers) of
+         (DontUseLedgerPeers, PeerSharingDisabled, DontUseBootstrapPeers)
+           -> NonPromiscuousMode
+         (DontUseLedgerPeers, PeerSharingDisabled, UseBootstrapPeers {})
+           |  Set.null knownBootstrapPeers
+           -> NonPromiscuousMode 
+         _ -> PromiscuousMode
```

Comment:
Isn't knowing about this mode only useful when in `TooOld` state? If so I don't think checking for `PeerSharingDisabled` or `DontUseLedgerPeers` is right. When in `TooOld` state we are only going to be connected to trusted peers, so the node can effectively have those flags enabled.

Reply:
But the way I understand it, this won't work in bootstrap mode, am I wrong?

Reply:
I just saw the logic above and I agree with it. However, I think that logic should be done in this function and not pollute the `peerSelectionGovernorLoop` It's still quite a large chunk of code.

Reply:
I added `outboundConnectionsState` function which compute the `OutboundConnectionsState`.